src/core/eggmath.rkt

```diff
@@ -0,0 +1,213 @@
+#lang racket
+
+(require ffi/unsafe
+         ffi/unsafe/define)
+(require racket/runtime-path)
+
+(require "../debug.rkt" "../common.rkt")
+(require "../syntax/rules.rkt")
+(require "../syntax/herbie-to-egg-pattern.rkt")
+
+(require egg-herbie)
+
+(module+ test (require rackunit))
+
+(provide egraph-run egraph-add-exprs egraph-run-iter egraph-get-simplest egg-expr->expr egg-add-exn? make-ffi-rules egraph-get-cost egraph-get-size)
```

Comment:
Generally I prefer to combine these together into just a few lines: Racket libraries; my libraries; other libraries; and `provide`. But also I'll usually split `provide` over multiple lines to avoid any longer than 72 characters.